homeassistant/components/onboarding/views.py

```diff
@@ -201,8 +203,15 @@ async def post(self, request, data):
                     "invalid client id or redirect uri", HTTP_BAD_REQUEST
                 )
 
+            if credential is None:
+                return self.json_message(
+                    "Credentials for user not available", HTTP_FORBIDDEN
+                )
+
             # Return authorization code so we can redirect user and log them in
-            auth_code = hass.components.auth.create_auth_code(data["client_id"], user)
+            auth_code = hass.components.auth.create_auth_code(
+                data["client_id"], credential
+            )
```

Comment:
This is the main use case of the global credential data in the request. We need the credential here since we create the refresh_token here. If we had some other way of remembering it from previous user step that could have been an alternative, but i think knowing the credential source of a request make sense.